src/MeasureToolbox/expectations.jl

```diff
@@ -212,17 +212,18 @@ end
 An efficient wrapper for [`expect`](@ref). Please see its doc string more
 information.
 """
-macro expect(expr, prefs, args...)
-    _error(str...) = InfiniteOpt._macro_error(:expect, (expr, prefs, args...), 
-                                              str...)
-    extra, kwargs, _, _ = InfiniteOpt._extract_kwargs(args)
-    if length(extra) > 0
-        _error("Unexpected positional arguments." *
-               "Must be of form @expect(expr, prefs, kwargs...).")
-    end
+macro expect(args...)
+    error_fn = InfiniteOpt.JuMPC.build_error_fn(:expect, args, __source__)
+    pos_args, kwargs = InfiniteOpt.JuMPC.parse_macro_arguments(
+        error_fn, 
+        args, 
+        num_positional_args = 2
+    )
+    expr, prefs = pos_args
     expression = MutableArithmetics.rewrite_and_return(expr)
```

Comment:
Separately, but consider using `rewrite_and_return(expr; move_factors_into_sums::Bool = false)` to opt-in to the new rewriting code. It doesn't do as many weird rewrites.

Reply:
Noted, I'll save that for another PR.